test/kt/esimd_radix_sort_utils.h

```diff
@@ -1,268 +1,296 @@
-    {
-        std::uniform_int_distribution<int> dist(std::numeric_limits<T>::lowest(), std::numeric_limits<T>::max());
-        std::generate(input, input + unique_threshold, [&] { return T(dist(gen)); });
-    }
-    else if constexpr (std::is_integral_v<T>)
-    {
-        std::uniform_int_distribution<T> dist(std::numeric_limits<T>::lowest(), std::numeric_limits<T>::max());
-        std::generate(input, input + unique_threshold, [&] { return dist(gen); });
-    }
-    else
-    {
-        std::uniform_real_distribution<T> dist_real(std::numeric_limits<T>::min(), log2(1e12));
-        std::uniform_int_distribution<int> dist_binary(0, 1);
-        auto randomly_signed_real = [&dist_real, &dist_binary, &gen]()
-        {
-            auto v = exp2(dist_real(gen));
-            return dist_binary(gen) == 0 ? v : -v;
-        };
-        std::generate(input, input + unique_threshold, [&] { return randomly_signed_real(); });
-    }
-    for (uint32_t i = 0, j = unique_threshold; j < size; ++i, ++j)
-    {
-        input[j] = input[i];
-    }
-}
-
-template <typename Container1, typename Container2>
-void
-print_data(const Container1& expected, const Container2& actual, std::size_t first, std::size_t n = 0)
-{
-    if (expected.size() <= first)
-        return;
-    if (n == 0 || expected.size() < first + n)
-        n = expected.size() - first;
-
-    if constexpr (std::is_floating_point_v<typename Container1::value_type>)
-        std::cout << std::hexfloat;
-    else
-        std::cout << std::hex;
-
-    for (std::size_t i = first; i < first + n; ++i)
-    {
-        std::cout << actual[i] << " --- " << expected[i] << std::endl;
-    }
-
-    if constexpr (std::is_floating_point_v<typename Container1::value_type>)
-        std::cout << std::defaultfloat << std::endl;
-    else
-        std::cout << std::dec << std::endl;
-}
-
-#endif // _ESIMD_RADIX_SORT_TEST_UTILS_H
+
+// -*- C++ -*-
+//===-- esimd_radix_sort_test_utils.h -----------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ESIMD_RADIX_SORT_TEST_UTILS_H
+#define _ESIMD_RADIX_SORT_TEST_UTILS_H
+
+#include <oneapi/dpl/experimental/kernel_templates>
+
+#if __has_include(<sycl/sycl.hpp>)
+#    include <sycl/sycl.hpp>
+#else
+#    include <CL/sycl.hpp>
+#endif
+
+#ifndef LOG_TEST_INFO
+#    define LOG_TEST_INFO 0
+#endif
+
+#include <string>
+#include <tuple>
+#include <random>
+#include <cmath>
+#include <limits>
+#include <iostream>
+#include <cstdint>
+#include <vector>
+#include <type_traits>
+
+#if defined(TEST_DATA_PER_WORK_ITEM) && defined(TEST_WORK_GROUP_SIZE)
+using ParamType = oneapi::dpl::experimental::kt::kernel_param<TEST_DATA_PER_WORK_ITEM, TEST_WORK_GROUP_SIZE>;
+constexpr ParamType kernel_parameters;
```

Comment:
What if `kernel_parameters` is undefined? Looks like we will have compile error...

Reply:
This solution mean that the Kernel template param now is the same for sort and sort by key algorithms.
Are you sure that this two algorithms still have the same structure of Kernel template params?

Reply:
> What if kernel_parameters is undefined? Looks like we will have compile error...

Currently, the tests are designed to strictly require `TEST_DATA_PER_WORK_ITEM` and `TEST_WORK_GROUP_SIZE` being externally defined. The dependency on these macros is a bit convoluted because of being scattered across multiple files. I can get rid of this alias to simplify the logic.

> This solution mean that the Kernel template param now is the same for sort and sort by key algorithms. Are you sure that this two algorithms still have the same structure of Kernel template params?

`oneapi::dpl::experimental::kt::kernel_param` is a genetic structure supposed to be used with any kernel template, be it esimd sort, other sort, scan and etc. The test harness passes different `TEST_DATA_PER_WORK_ITEM` and `TEST_WORK_GROUP_SIZE` when compiling different tests. So there is no dependency between `radix_sort` and `radix_sort_by_key` algorithms on the same `TEST_DATA_PER_WORK_ITEM` and `TEST_WORK_GROUP_SIZE` configuration.

Reply:
ok